### PR TITLE
Improve error messages for `match?` assertions

### PR DESCRIPTION
**Pull request by lexmag**
Follow-up to https://github.com/elixir-lang/elixir/issues/3488#issuecomment-133423879.
* [x] `assert match?(..)`
* [x] `refute match?(..)`

### CODE REVIEW

**Comment by lexmag** [comment]
@josevalim this is ready for review.

**Comment by josevalim** [comment]
Let's add a test to `assertion_test.exs` please!

**Comment by lexmag** [comment]
Added.

File: lib/ex_unit/lib/ex_unit/assertions.ex [hunk 1]
```diff
@@ -130,6 +141,17 @@ defmodule ExUnit.Assertions do
       refute age < 0
 
   """
+  defmacro refute({:match?, _, [left, right]} = assertion) do
+    code = Macro.escape(assertion)
+    quote do
+      right = unquote(right)
+      not assert not(match?(unquote(left), right)),
```
Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/assertions.ex]: The first `not` makes a warning appear:
```
test/elixir/kernel/binary_test.exs:95: warning: the result of the expression is ignored (suppress the warning by assigning the expression to the _ variable)
```

Comment (lexmag) [inline lib/ex_unit/lib/ex_unit/assertions.ex]: It's resolved by using `refute/2`.

**Pull request merged**